sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/CompactibleFileStreamLog.scala

```diff
@@ -112,7 +112,16 @@ abstract class CompactibleFileStreamLog[T <: AnyRef : ClassTag](
    * Default implementation retains all log entries. Implementations should override the method
    * to change the behavior.
    */
-  def shouldRetain(log: T): Boolean = true
+  def shouldRetain(log: T, context: Map[String, Any]): Boolean = true
```

Comment:
I feel adding `context` is overkill. How about passing the now timestamp into this instead? We can add the retain context like this in future if that's necessary:

```
def shouldRetain(log: T, currentTime: Long): Boolean = true

def shouldRetain(log: T, context: Map[String, Any]): Boolean = {
  shouldRetain(log, context.get("currentTime").asInstanceOf[Long])
}
```

Reply:
Actually the change was made to cover possible general case, because I got review comment when making change of shouldRetain that "please avoid breaking backward compatibility." I disagreed as it's a private API based on the package, but took considerable time and effort to persuade.

If we agree that this is not a public API and shouldn't bother with backward compatibility with this, I definitely agree this is pretty much overkill as of now. WDYT?

Reply:
Yep, this is not a public API. It's okay to change it.

Reply:
Thanks! Will make a quick fix.

Reply:
Fixed. Please take a look again. Thanks!